x-pack/plugins/cases/public/components/create/form_context.test.tsx

```diff
@@ -780,4 +786,58 @@ describe('Create case', () => {
       expect(screen.queryByTestId('createCaseAssigneesComboBox')).toBeNull();
     });
   });
+
+  describe('draft comment', () => {
+    describe('existing storage key', () => {
+      beforeEach(() => {
+        sessionStorage.setItem(defaultCreateCaseForm.draftStorageKey, 'value set in storage');
+      });
+
+      afterEach(() => {
+        sessionStorage.removeItem(defaultCreateCaseForm.draftStorageKey);
+      });
+
+      it('should have session storage value same as draft comment', async () => {
+        mockedContext.render(
+          <FormContext onSuccess={onFormSubmitSuccess}>
+            <CreateCaseFormFields {...defaultCreateCaseForm} />
+            <SubmitCaseButton />
+          </FormContext>
+        );
+
+        await waitForFormToRender(screen);
+        const descriptionInput = within(screen.getByTestId('caseDescription')).getByTestId(
+          'euiMarkdownEditorTextArea'
+        );
+
+        expect(descriptionInput).toHaveValue('value set in storage');
+      });
+    });
+
+    describe('set storage key', () => {
+      afterEach(() => {
+        sessionStorage.removeItem(defaultCreateCaseForm.draftStorageKey);
+      });
+
+      it('should have session storage value same as draft comment', async () => {
```

Comment:
Shouldn't we use `jest.advanceTimersByTime(1000);` here?

Reply:
I added it initially, but removing it also passed the test as `await waitForFormToRender(screen);` seems to wait long enough to set the sessionStorage value in description .

Reply:
I think relying on `jest.advanceTimersByTime` is better because we have total control of "time" in the debounce call. It is ok to leave it as you are right, `await waitForFormToRender(screen);` seems to wait for enough. I am not sure though if in the future we refactor the form and the `await waitForFormToRender(screen);` becomes faster what will happen to the test. We should see then I guess 🙂

Reply:
yeah, there is a chance of it failing if `await waitForFormToRender(screen);` becomes faster. Let me add `jest.advanceTimersByTime` back just in case 😄